staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter.go

```diff
@@ -117,10 +115,10 @@ func (c *converterImpl) FromUnstructured(u map[string]interface{}, obj interface
 		newObj := reflect.New(t.Elem()).Interface()
 		newErr := fromUnstructuredViaJSON(u, newObj)
 		if (err != nil) != (newErr != nil) {
-			glog.Fatalf("FromUnstructured unexpected error for %v: error: %v", u, err)
+			panic(fmt.Errorf("FromUnstructured unexpected error for %v: error: %v / %v", u, err, newErr))
```

Comment:
Good to know, thanks. However I think panics should probably be caught at the top of the stack and then logged in a desired way - I think this is how it works already? This allows to decouple two concerns - logging and failing assertions+panics from bugs.